### DISCUSSION TITLE
Welcome to APE Discussions!

### DISCUSSION THREAD

**Discussion by Kaljurand** [Attempto/APE#36] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lbrock** [comment]
I'm an independent developer, using parts of Attempo to explore symbolic AI. 
Installed all the components of ACE (except the reasoner) on Amazon Web a few years ago, and did get it working.
Have accumulated several questions:
- was the Attempo reasoner (RACE, I think) ever released, or is it available somewhere?
- Has anybody connected Allegrograph (from Franz Inc) as a reasoner for Attempto?
- Is there any collection of English prose translated ACE?
- Are there any other "lexicons" for ACE other than the two on the site?

**Reply by nefuchs** [reply]
Hi lbrock

**was the Attempto reasoner (RACE, I think) ever released, or is it available somewhere?**
RACE will be on GitHub "real soon now". In the meantime please rely on its web-interface and its web-service.

**Has anybody connected Allegrograph (from Franz Inc) as a reasoner for Attempto?**
I am not aware of this. If you are looking for an alternative to RACE, ACE can be used as input to the Automated Theorem Prover TPTP (https://www.tptp.org).

   --- nef

**Reply by nivekuil** [reply]
also interested in RACE being released

**Reply by ExternalReality** [reply]
I would like RACE to be released as well.

**Reply by tkuhn** [reply]
RACE has been released in the meantime: https://github.com/Attempto/RACE

**Reply by hurufu** [reply]
Thanks for release! But I get error 404, probably it is still private repo.

**Reply by tkuhn** [reply]
Right, it wasn't released yet, but now it is! You can find it here: https://github.com/Attempto/RACE

**Comment by hurufu** [comment]
Hello,
I'm also interested in Attempto project. As an exercise in Prolog programming I have written a _(not finished)_ [expert system](https://github.com/hurufu/ffmpeg-planner) with human interface in ACE.

The main purpose of it is to help users to use `ffmpeg` program without reading tons of documentation. If a non sophisticated ffmpeg user tries to do any non-trivial (or even trivial) video conversions he will find that command line arguments are very difficult to grasp. So I wanted to create a system in which you describe in ACE what do you want in your output video file (eg. 2nd audio track removed, video resized and subtitles included from an external file), and the expert system using PDDL planner should decide on which command line switches to ffmpeg should you use in order to transform your inputs files into desired output file. It can even perform the conversion for you.
It is very far from a finished program and I don't have much time to work on it, but one day I might try to revisit this project.

**Reply by hurufu** [reply]
Also, completely forgot. I've played with ACE for formalizing manual test cases, but sadly without any particular success, because even small things require quite a lot of text.

A small excerpt from a test case looked as follows:

```
If p:Test-Stage >= 2 then p:DUT performs p:Contactless-Combination-Selection.
If p:Test-Stage >= 2 then p:Mutually-Supported-Application-List has all applications that p:Test-Card-1 has.
If p:Test-Stage >= 3 then p:DUT presents all applications that p:Mutually-Supported-Application-List has.
If p:Test-Stage >= 4 then p:Tester chooses p:Visa-1.
If p:Test-Stage >= 5 then "Approved" is an applicable message.
If p:Test-Stage >= 5 then "Declined" is an inapplicable message.
```

 For example a particularly hard to define was one of the simplest requirements: to define a notion of a bit to check if a particular bit was set in the outcome of a program. I wanted to have something akin to "the first bit of a second byte is set",
 but ended up with `X has "[2,1]"`.

**Comment by lbrock** [comment]
I’m also interested – still.   Checked https://github.com/Attempto/, but not listed there.
